src/daemon/rpmostreed-utils.cxx

```diff
@@ -609,3 +609,42 @@ rpmostreed_parse_revision (const char  *revision,
 out:
   return ret;
 }
+
+/* Throw an error if there exists systemd inhibitor locks in `block` mode only.
+ * Note: systemd 248 provides a `--check-inhibitors` option, but it also checks
+ * for inhibitors in `delay` mode, which isn't what we want. */
+gboolean
+check_sd_inhibitor_locks (GDBusConnection *connection,
+                          GCancellable    *cancellable,
+                          GError         **error)
+{
+  g_autoptr(GVariant) inhibitors_array_tuple = 
+    g_dbus_connection_call_sync (connection, "org.freedesktop.login1", "/org/freedesktop/login1",
```

Comment:
My second guess would be perhaps we shouldn't block the daemon with this synchronous D-Bus call?